### PR TITLE
Adds some visual effects when ammo casings are fired and when projectiles hit things

### PR DESCRIPTION
**Pull request by phil235**
:cl: phil235
add: Added new things
imageadd: added some icons and images
/:cl:

gunviz

### CODE REVIEW

**Comment by PKPenguin321** [comment]
pics

File: code/game/objects/effects/overlays.dm [hunk 1]
```diff
@@ -55,12 +55,13 @@
 	duration = 5
 	randomdir = FALSE
 	layer = BELOW_MOB_LAYER
+	var/splatter_type = "splatter"
```
Comment (ChangelingRain) [inline code/game/objects/effects/overlays.dm]: :+1:
might want to rename the path, though

**Comment by RemieRichards** [comment]
tl;dr:
* support for muzzle flashes/effects when a bullet is fired
* mobs make a little blood splatter when they're hit (xenos have a sprite for this too)

File: code/modules/projectiles/firing.dm [hunk 2]
```diff
@@ -40,6 +40,13 @@
 	if (!istype(targloc) || !istype(curloc) || !BB)
 		return 0
 	BB.ammo_casing = src
+
+	var/firing_dir
+	if(BB.firer)
+		firing_dir = BB.firer.dir
+	if(!BB.suppressed && firing_effect_type)
+		new firing_effect_type(get_turf(src), firing_dir)
```
Comment (ChangelingRain) [inline code/modules/projectiles/firing.dm]: `PoolOrNew(firing_effect_type, list(get_turf(src), firing_dir))` though I still have no idea if it's performance efficient.

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -75,10 +76,23 @@
 		return "chest"
 
 /obj/item/projectile/proc/on_hit(atom/target, blocked = 0)
+	var/turf/target_loca = get_turf(target)
 	if(!isliving(target))
+		if(impact_effect_type)
+			new impact_effect_type (target_loca, target, src)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile.dm]: `PoolOrNew(impact_effect_type, list(target_loca, target, src))` though see above

File: code/modules/projectiles/projectile.dm [hunk 4]
```diff
@@ -75,10 +76,23 @@
...
 	if(blocked != 100) // not completely blocked
+		if(damage && L.blood_volume && damage_type == BRUTE)
+			if(isalien(L))
+				new /obj/effect/overlay/temp/dir_setting/bloodsplatter/xenosplatter(target_loca, dir)
+			else
+				new /obj/effect/overlay/temp/dir_setting/bloodsplatter(target_loca, dir)
+			if(prob(33))
+				L.add_splatter_floor(target_loca)
+		else if(impact_effect_type)
+			new impact_effect_type (target_loca, target, src)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile.dm]: etc etc you get the idea

**Comment by WJohn** [comment]
More visual feedback is always good :+1:

**Comment by MrPerson** [comment]
Great changelog entry you got there

Neato effect

**Comment by Tacolizard** [comment]
goon lighting when

**Review by ChangelingRain — COMMENTED** [review]
Oh hey this is new, summary;
AAAA USE PoolOrNew() AAAA

File: code/game/objects/effects/overlays.dm [hunk 5]
```diff
@@ -90,44 +91,37 @@
...
 
-/obj/effect/overlay/temp/heal //color is white by default, set to whatever is needed
-	name = "healing glow"
-	icon_state = "heal"
-	duration = 15
+/obj/effect/overlay/temp/dir_setting/firing_effect
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "firing_effect"
+	mouse_opacity = 0
+	density = 0
```
Comment (ChangelingRain) [inline code/game/objects/effects/overlays.dm]: Both this and the above var are already set like this on the parent type.

File: code/game/objects/effects/overlays.dm [hunk 6]
```diff
@@ -393,6 +387,82 @@
...
+	icon_state = "blob_attack"
+	alpha = 140
+	randomdir = 0
+	duration = 6
+
+/obj/effect/overlay/temp/impact_effect
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "impact_bullet"
+	mouse_opacity = 0
+	density = 0
```
Comment (ChangelingRain) [inline code/game/objects/effects/overlays.dm]: Both this and the above var are already set like this on the parent type.

**Review by ChangelingRain — CHANGES_REQUESTED** [review]
Use the inheritance phil

**Review by Razharas — APPROVED** [review]
Im Razharas and i approve this message

**Comment by GunHog** [comment]
This is cooler than cool!

**Comment by phil235** [comment]
Alright, I did the requested changes.

**Review by ChangelingRain — APPROVED** [review]

**Pull request merged**